### PR TITLE
Adding function for client side

### PR DESCRIPTION
**Pull request by pinghaoluo**

### CODE REVIEW

File: src/cli/cli_parser.cc [hunk 1]
```diff
@@ -137,3 +138,25 @@ std::string CliParser::ParseMonitor(const std::string& uname) {
 
   return "Monitor complete";
 }
+
+std::string CliParser::ParseStream(const std::string& uname) {
+  if (!(FLAGS_reg.empty() && FLAGS_chirp.empty() && FLAGS_reply.empty() &&
+        FLAGS_read.empty() && FLAGS_follow.empty())) {
```
Comment (allenyhu) [inline src/cli/cli_parser.cc]: add check for FLAGS_monitor.empty() here

File: src/cli/cli_parser.cc [hunk 2]
```diff
@@ -137,3 +138,25 @@ std::string CliParser::ParseMonitor(const std::string& uname) {
...
+    return "Cannot Register, Reply, Read, or Follow with Monitor.";
+  }
+
+  if (uname.empty()) {
```
Comment (allenyhu) [inline src/cli/cli_parser.cc:158]: Do you actually need to be logged in to stream? @barath

File: src/cli/cli_parser.cc [hunk 3]
```diff
@@ -137,3 +138,25 @@ std::string CliParser::ParseMonitor(const std::string& uname) {
...
+    return "Must be logged in to perform actions.";
+  }
+
+  while (true) {
+    std::vector<ChirpObj> chirps = service_.Monitor(uname);
```
Comment (allenyhu) [inline src/cli/cli_parser.cc]: Why is this calling Monitor?

File: src/cli/cli_parser.cc (hunk 2 above)
Comment (barath) [reply src/cli/cli_parser.cc:158]: There isn't a real need, but the way I wrote the spec I think specifies that you need to be logged in for any action.

**Review by allenyhu — APPROVED** [review]

**Pull request closed without merging**